File: de.fu_berlin.inf.dpp.ui/src/de/fu_berlin/inf/dpp/ui/browser_functions/ShowSessionWizard.java
package de.fu_berlin.inf.dpp.ui.browser_functions;

import de.fu_berlin.inf.ag_se.browser.functions.JavascriptFunction;
import de.fu_berlin.inf.dpp.HTMLUIContextFactory;
import de.fu_berlin.inf.dpp.ui.ide_embedding.DialogManager;
import de.fu_berlin.inf.dpp.ui.pages.SessionWizardPage;

/**
 * Offers a via Javascript invokable method to open and display the
 * {@link SessionWizardPage} dialog.
 * <p>
 * JS-signature: "void __java_showSessionWizard()".
 */
public class ShowSessionWizard extends JavascriptFunction {
    private DialogManager dialogManager;
    private SessionWizardPage sessionWizardPage;
    public static final String JS_NAME = "showSessionWizard";

    // TODO: Rename to openXYZ for more convenient naming

    /**
     * Created by PicoContainer
     * 
     * @param dialogManager
     * @param sessionWizardPage
     * @see HTMLUIContextFactory
     */
    public ShowSessionWizard(DialogManager dialogManager,
        SessionWizardPage sessionWizardPage) {
        super(NameCreator.getConventionName(JS_NAME));
        this.dialogManager = dialogManager;
        this.sessionWizardPage = sessionWizardPage;

    }

    @Override
    public Object function(Object[] arguments) {
        dialogManager.showDialogWindow(sessionWizardPage);
        return null;
    }

}

File: de.fu_berlin.inf.dpp.ui/src/de/fu_berlin/inf/dpp/ui/browser_functions/ShowSessionWizard.kt
package de.fu_berlin.inf.dpp.ui.browser_functions

import de.fu_berlin.inf.dpp.HTMLUIContextFactory
import de.fu_berlin.inf.dpp.ui.ide_embedding.DialogManager
import de.fu_berlin.inf.dpp.ui.pages.SessionWizardPage
import de.fu_berlin.inf.dpp.ui.pages.IBrowserPage

/**
 * Open and display the {@link SessionWizardPage} dialog.
 * Created by PicoContainer
 *
 * @param dialogManager
 * @param sessionWizardPage
 * @see HTMLUIContextFactory
 */
class ShowSessionWizard ( dialogManager: DialogManager?, sessionWizardPage: SessionWizardPage?)
	 					: TypedJavaScriptFunction(JS_NAME) {
	private val dialogManager: DialogManager? = dialogManager
	private val sessionWizardPage : IBrowserPage? = sessionWizardPage

	/**
	 * Open and display the {@link SessionWizardPage} dialog.
	 */
	@BrowserFunction
	fun showSessionWizard() {
		dialogManager!!.showDialogWindow(sessionWizardPage)
	}

	companion object {
		// TODO: Rename to openXYZ for more convenient naming
		val JS_NAME = "showSessionWizard"
	}
}